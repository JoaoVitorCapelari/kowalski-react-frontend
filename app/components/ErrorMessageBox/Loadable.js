/**
 *
 * Asynchronously loads the component for ErrorMessageBox
 *
 */

import Loadable from 'react-loadable';

export default Loadable({
  loader: () => import('./index'),
  loading: () => null,
});
