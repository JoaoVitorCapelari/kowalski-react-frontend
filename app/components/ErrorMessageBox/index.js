/**
*
* ErrorMessageBox
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Container = styled.div`
  display: flex;
  flex-grow: 1;
`;

function ErrorMessageBox({ errorMsg }) {
  if (!errorMsg) {
    return <div></div>;
  }

  return (
    <Container className="message is-danger">
      <Container className="message-body">
        { errorMsg }
      </Container>
    </Container>
  );
}

ErrorMessageBox.propTypes = {
  errorMsg: PropTypes.string,
};

export default ErrorMessageBox;
