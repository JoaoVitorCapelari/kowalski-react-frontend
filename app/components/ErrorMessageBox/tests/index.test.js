import React from 'react';
import { shallow } from 'enzyme';

import ErrorMessageBox from '../index';

describe('<ErrorMessageBox />', () => {
  it('renders properly when there are no messages', () => {
    const renderedComponent = shallow(<ErrorMessageBox />);
    expect(renderedComponent.html()).toMatchSnapshot();
  });

  it('renders properly when there is a message', () => {
    const renderedComponent = shallow(<ErrorMessageBox errorMsg="test" />);
    expect(renderedComponent.html()).toMatchSnapshot();
  });
});
