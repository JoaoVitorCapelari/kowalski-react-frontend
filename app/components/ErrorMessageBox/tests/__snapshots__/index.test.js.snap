// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessageBox /> renders properly when there are no messages 1`] = `"<div></div>"`;

exports[`<ErrorMessageBox /> renders properly when there is a message 1`] = `"<div class=\\"message is-danger ErrorMessageBox__Container-dsKmoD jLVSwp\\"><div class=\\"message-body ErrorMessageBox__Container-dsKmoD jLVSwp\\">test</div></div>"`;
